### DISCUSSION TITLE
blog/math-and-python-implementation-for-multi-class-logistic-regression/

### DISCUSSION THREAD

**Discussion by giscus** [jeremy-feng/giscus#11] [Announcements]
# blog/math-and-python-implementation-for-multi-class-logistic-regression/

多元 Logistic 回归的目标函数推导，并用 Python 实现 OvO、OvR 和直接构造多元 Logistic 模型的方法。

https://jeremy-feng.github.io/blog/math-and-python-implementation-for-multi-class-logistic-regression/

**Comment by Flyme-wang** [comment]
博主，您方便分享一下这个博文中的数据吗

**Reply by jeremy-feng** [reply]
你好，这是数据文件：

[letter_recognition.csv](https://github.com/jeremy-feng/giscus/files/15084667/letter_recognition.csv)

**Reply by Flyme-wang** [reply]
非常感谢博主！

**Comment by wangge58** [comment]
博主你好，我想问下如果想要预测一个有13列x和1列y的数据的y（最后一列）的值，上面的代码应该改的地方有哪些

**Reply by jeremy-feng** [reply]
只需要把读取数据的代码修改一下即可。如果你的 x 在左边 13 列，y 在最后一列，可以这样进行切片：

```python
X = data.iloc[:, :13]
y = data.iloc[:, -1]
```
